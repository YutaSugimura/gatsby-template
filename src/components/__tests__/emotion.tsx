import React from 'react';
import renderer from 'react-test-renderer';
import serializer from 'jest-emotion';
import SimpleEmotion from '../emotionComponents/simple';
import WithButton from '../emotionComponents/withEmotion';

expect.addSnapshotSerializer(serializer);

test('renders with correct styles', () => {
  const tree = renderer
    .create(<WithButton>emotion button</WithButton>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders component styles', () => {
  const tree = renderer.create(<SimpleEmotion />).toJSON();
  expect(tree).toMatchSnapshot();
});
