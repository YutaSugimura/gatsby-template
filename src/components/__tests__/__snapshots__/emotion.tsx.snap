// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component styles 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "3h4kzc",
      "next": undefined,
      "styles": "
  color: hotpink;
",
      "toString": [Function],
    }
  }
>
  Some hotpink text.
  <div
    css={
      Object {
        "map": undefined,
        "name": "1081t4c",
        "next": undefined,
        "styles": "text-decoration:underline;",
        "toString": [Function],
      }
    }
  >
    Some text
  </div>
</div>
`;

exports[`renders with correct styles 1`] = `
.emotion-0 {
  color: blue;
}

<button
  className="emotion-0"
>
  emotion button
</button>
`;
